smarts/core/utils/logging.py

```diff
@@ -29,3 +34,50 @@ def timeit(name: str):
     ellapsed_time = (time() - start) * 1000
 
     logging.info(f'"{name}" took: {ellapsed_time:4f}ms')
+
+
+libc = ctypes.CDLL(None)
+try:
+    c_stdout = ctypes.c_void_p.in_dll(libc, "stdout")
+except:
+    # macOS
+    c_stdout = ctypes.c_void_p.in_dll(libc, "__stdoutp")
+
+
+def try_fsync(fd):
+    try:
+        os.fsync(fd)
+    except OSError:
+        # On GH actions was returning an OSError: [Errno 22] Invalid argument
+        pass
```

Comment:
Not completely comfortable with skipping all `OSError`s here can we filter just for `[Errno 22]`?

Reply:
Why not? if the fsync fails, it's not the end of the world.

Reply:
That is fair.